cdap-ui/app/cdap/components/MarketActionsContainer/index.js

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+import React, {Component, PropTypes} from 'react';
+import T from 'i18n-react';
+import getIcon from 'services/market-action-icon-map';
+import shortid from 'shortid';
+import classnames from 'classnames';
+import AbstractWizard from 'components/AbstractWizard';
+
+require('./MarketActionsContainer.less');
+
+export default class MarketActionsContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      completedActions: [],
+      wizard: {
+        actionIndex: null,
+        actionType: null,
+        action: null
+      }
+    };
+  }
+
+  closeWizard(returnResult) {
+    if (returnResult) {
+      this.setState({
+        completedActions: this.state.completedActions.concat([this.state.wizard.actionIndex]),
+        wizard: {
+          actionIndex: null,
+          actionType: null,
+          action: null
+        }
+      });
+      return;
+    }
+    this.setState({
+      wizard: {
+        actionIndex: null,
+        actionType: null
+      }
+    });
+  }
+
+  openWizard(actionIndex, actionType, action) {
+    this.setState({
+      wizard: {
+        actionIndex,
+        actionType,
+        action
+      }
+    });
+  }
+
+  render() {
+    if (!Array.isArray(this.props.actions)) {
+      return null;
+    }
+    return (
+      <div className="market-entity-actions">
+        {
+          this.props
+            .actions
+            .map((action, index) => {
+              let isCompletedAction = this.state.completedActions.indexOf(index) !== -1 ;
+              let actionName = T.translate('features.Market.action-types.' + action.type + '.name');
+              let actionIcon = getIcon(action.type);
+              return (
+                <div
+                  className="action-container text-center"
+                  key={shortid.generate()}
```

Comment:
Will this cause a problem by re-rendering all the child element every time the render() is called?